### ISSUE TITLE
Use X509 certificate for corporate login

### ISSUE THREAD

**Issue by nicolamori** [IsmaelMartinez/teams-for-linux#1030]
**Is your feature request related to a problem? Please describe.**
Every time I open the client I have to type my username&password credentials into my company's login page. Since I use a rather involved password (which is usually remembered by the browser when I browse other sites of my company) I have to retrieve it from my vault every time. This is a quite inconvenient workflow for me.

**Describe the solution you'd like**
Since my company's SSO service allows the usage of X509 certificates it would be much more handy for me to install my personal certificate in Teams For Linux and log in with that.

**Describe alternatives you've considered**
I have gnome-keyring and kwallet installed, but still my company's login page as displayed in Teams For Linux does not remember my username&password credentials. So I thought that using X509 could be a viable solution to my problem.

**Comment by IsmaelMartinez** [comment]
have you tried the customCACertsFingerprints, clientCertPath and clientCertPassword config option in https://github.com/IsmaelMartinez/teams-for-linux/tree/develop/app/config ? 

Info in https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/app/certificate/README.md

I am not sure if that would work but maybe?

**Comment by nicolamori** [comment]
I didn't know about those options. Specifying the full path to the certificate file with `clientCertPath` and the certificate password with `clientCertPassword` everything works beautifully: on my corporate login page I choose to identify myself with a certificate and then I'm logged in flawlessly without having to type anything.

What I don;t understand is that now if I log out, clear storage on exit, launch again teams-for-linux without any of the above options then the certificate login seems to still work. I revoked the kwallet permission for teams-for-linux but still login with certificate works. I even deleted the certificate file and it still works... I think it should have been cached somewhere but I don't understand where.

Despite these details I'd say that login with certificate definitely works, so thank you very much for your support.

**Comment by IsmaelMartinez** [comment]
that is odd. have you tried removing the cache files described in here? https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/KNOWN_ISSUES.md#blank-page

Or all the `.config/teams-for-linux/` . In theory electron do not store cache outside of that folder.

**Comment by nicolamori** [comment]
Well, I started the application, logged out, quit with `Clear storage`, removed `.config/teams-for-linux/`, revoked all the application permissions in kwallet and disabled kwallet system-wise, deleted anything in gnome-keyring with seahorse, started again the applications without any certificate option, and still I was able to log in successfully with the certificate. I can reproduce this behavior consistently. I'm really clueless about what's going on with this certificate.

**Comment by nicolamori** [comment]
Could it be that the certificate has been stored in the Chromium vault? I have Chromium installed as my second browser, I rarely use it, and I noticed that the certificate is available in there. As far as I remember I never installed the certificate in chromium, so could it be that when I launched teams-for-linux with certificate options it correctly retrieved the certificate and then stored in the Chromium vault, and now it uses it from there?
